python/cudf/tests/test_dataframe.py

```diff
@@ -176,6 +176,24 @@ def test_series_init_none():
     # values should match despite whitespace difference
     assert got.split() == expect.split()
 
+@pytest.mark.parametrize('nan_as_null, dropna', 
+                        [(True, True), (True, False), 
+                         (False, True), (False, False)])
+def test_series_nunique(nan_as_null, dropna):
+    nan = np.dtype('float64').type(np.nan)
+    pd_series = pd.Series([1.0, 2.0, 3.0, -1.0, nan])
+    cudf_series = Series([1.0, 2.0, 3.0, nan, None], nan_as_null=nan_as_null)
```

Comment:
Add a comment here that explains that we remove nulls as opposed to NaNs, so for testing against Pandas we need to replace NaN with another discrete value.